File: mobile_thru_project_server_full/src/main/java/com/ssafy/cafe/model/service/AdminService.java
package com.ssafy.cafe.model.service;

import java.util.List;
import java.util.Map;

import com.ssafy.cafe.model.dto.Admin;
import com.ssafy.cafe.model.dto.Order;
import com.ssafy.cafe.model.dto.User;

public interface AdminService {

	
	// 조회된 관리자 정보를 반환한다.
    public Admin login(String a_id, String a_pass);
    
    // 어드민 정보를 갱신한다.
    public int update(Admin admin);
    
    // store id에 해당하는 admin의 토큰 값을 반환한다.
    public Map selectAdmin(String a_id);
}

File: android_admin/mobile_cafe_admin_fianl_project/app/src/main/java/com/ssafy/mobile_cafe_admin_fianl_project/src/main/service/AdminService.kt
package com.ssafy.mobile_cafe_admin_fianl_project.src.main.service

import android.util.Log
import com.ssafy.mobile_cafe_admin_fianl_project.src.main.dto.Admin
import com.ssafy.mobile_cafe_admin_fianl_project.util.RetrofitCallback
import com.ssafy.mobile_cafe_admin_fianl_project.util.RetrofitUtil
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class AdminService {

    // 관리자 로그인
    fun login(admin: Admin, callback: RetrofitCallback<Admin>) {
        RetrofitUtil.adminService.login(admin).enqueue(object: Callback<Admin> {
            override fun onResponse(call: Call<Admin>, response: Response<Admin>) {
               val res = response.body()
                if(response.code() == 200) {
                    if(res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                        callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Admin>, t: Throwable) {
                callback.onError(t)
            }

        })
    }

    fun update(admin: Admin) {
        RetrofitUtil.adminService.update(admin).enqueue(object : Callback<Int> {
            override fun onResponse(call: Call<Int>, response: Response<Int>) {
                val res = response.body()
                Log.d(TAG, "update: $admin")
                Log.d(TAG, "update: ${response.code()}")
                if (response.isSuccessful) {
                    Log.d(TAG, "onResponse: success")
                } else {
                    Log.d(TAG, "onResponse: Error Code : ${response.code()}")
                }
            }

            override fun onFailure(call: Call<Int>, t: Throwable) {
                Log.d(TAG, "onFailure: $t")
            }
        })
    }

}